src/store/modules/keymap.js

```diff
@@ -238,6 +237,31 @@ const mutations = {
     }
     state.selectedIndex = index;
   },
+  /**
+   * Use this after switching out the keycodes for a new layout to keep the keymap
+   * in sync. It has it's own copy of the labels, which need to be updated for
+   * correct ISO support.
+   * @param {} state
+   */
+  updateKeycodeNames(state) {
+    let store = this;
+    // assumes the keycode store has changed due to layout update
+    state.keymap = state.keymap.reduce((layers, layer) => {
```

Comment:
to ensure reactivity works correctly, just change the reference to a new keymap with all the old values copied into it. This will force the UI to re-render.